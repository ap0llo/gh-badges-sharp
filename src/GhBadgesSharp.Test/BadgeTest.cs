using System.Linq;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Xunit;

namespace GhBadgesSharp.Test
{   
    [UseReporter(typeof(DiffReporter))]
    public partial class BadgeTest
    {
        private class ApprovalNamer : UnitTestFrameworkNamer
        {
            private readonly int m_Id;

            public override string Name => $"{base.Name}_{m_Id:000}";

            public ApprovalNamer(int id)
            {
                m_Id = id;
            }

            public override string GetSubdirectory() => "/testdata";
        }


        [Theory]
        //          id  template                leftText   rightText  color       labelColor    logo    logoPosition    logoWidth    link1                  link2
        [InlineData(1,  BadgeStyle.Flat,        "Hello",   "World",   "yellow",   null,         null,   null,           null,        null,                  null)]
        [InlineData(2,  BadgeStyle.Flat,        "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  null)]
        [InlineData(3,  BadgeStyle.Plastic,     "Hello",   "World",   "yellow",   null,         null,   null,           null,        null,                  null)]
        [InlineData(4,  BadgeStyle.Plastic,     "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  null)]
        [InlineData(5,  BadgeStyle.Flat,        "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  "https://www.github.com")]
        [InlineData(6,  BadgeStyle.Plastic,     "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  "https://www.github.com")]
        [InlineData(7,  BadgeStyle.Plastic,     "Hello",   "World",   "yellow",   null,         null,   null,           null,        "",                    null)]
        [InlineData(8,  BadgeStyle.Flat,        "Hello",   "World",   "yellow",   null,         null,   null,           null,        "",                    null)]
        [InlineData(9,  BadgeStyle.Flat,        "Hello",   "World",   "red",      "blue",       null,   null,           null,        null,                  null)]
        [InlineData(10, BadgeStyle.Plastic,     "Hello",   "World",   "red",      "blue",       null,   null,           null,        null,                  null)]
        [InlineData(11, BadgeStyle.FlatSquare,  "Hello",   "World",   "yellow",   null,         null,   null,           null,        null,                  null)]
        [InlineData(12, BadgeStyle.FlatSquare,  "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  null)]
        [InlineData(13, BadgeStyle.FlatSquare,  "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  "https://www.github.com")]
        [InlineData(14, BadgeStyle.FlatSquare,  "Hello",   "World",   "yellow",   null,         null,   null,           null,        "",                    null)]
        [InlineData(15, BadgeStyle.FlatSquare,  "Hello",   "World",   "red",      "blue",       null,   null,           null,        null,                  null)]
        [InlineData(16, BadgeStyle.ForTheBadge, "Hello",   "World",   "yellow",   null,         null,   null,           null,        null,                  null)]
        [InlineData(17, BadgeStyle.ForTheBadge, "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  null)]
        [InlineData(18, BadgeStyle.ForTheBadge, "Hello",   "World",   "yellow",   null,         null,   null,           null,        "http://example.com",  "https://www.github.com")]
        [InlineData(19, BadgeStyle.ForTheBadge, "Hello",   "World",   "yellow",   null,         null,   null,           null,        "",                    null)]
        [InlineData(20, BadgeStyle.ForTheBadge, "Hello",   "World",   "red",      "blue",       null,   null,           null,        null,                  null)]
        public void MakeBadge_returns_expected_svg(int id, BadgeStyle style, string leftText, string rightText, string color, string labelColor, string logo, int? logoPosition, int? logoWidth, string link1, string link2)
        {
            var badge = Badge.MakeBadge(style, leftText, rightText, color, labelColor, logo, logoPosition, logoWidth, link1, link2);
            var writer = new ApprovalTextWriter(badge.ToString(), "svg");

            Approvals.Verify(writer, new ApprovalNamer(id), Approvals.GetReporter());            
        }   

    }
}
