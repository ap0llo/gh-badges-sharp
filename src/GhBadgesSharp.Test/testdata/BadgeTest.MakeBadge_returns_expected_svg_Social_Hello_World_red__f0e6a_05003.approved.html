<!DOCTYPE html>
            <html>
            <head>
                <style>
                body {
                    font-family: Arial, Helvetica, sans-serif;
                    font-size: 80%;
                }
                h1 {
                    font-size: 14pt;
                }
                thead {
                    font-weight: bold
                }
                table {
                    border-collapse: collapse;
                }
                table, th, td {
                    border: 1px solid black;
                }
                td {
                    padding: 0 10px 0 5px;
                }
                </style>
            </head>

            <body>

            <h1>Rendered Badge:</h1>
            <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="82.652" height="20">
  <style type="text/css"><![CDATA[
      a #llink:hover { fill:url(#b); stroke:#ccc; }
      a #rlink:hover { fill:#4183C4; }
    ]]></style>
  <linearGradient id="a" x2="0" y2="100%">
    <stop offset="0" stop-color="#fcfcfc" stop-opacity="0" />
    <stop offset="1" stop-opacity=".1" />
  </linearGradient>
  <linearGradient id="b" x2="0" y2="100%">
    <stop offset="0" stop-color="#ccc" stop-opacity=".1" />
    <stop offset="1" stop-opacity=".1" />
  </linearGradient>
  <g stroke="#d5d5d5">
    <rect stroke="none" fill="#fcfcfc" x="0.5" y="0.5" width="37.533" height="19" rx="2" />
    <rect y="0.5" x="44.033" width="38.119" height="19" rx="2" fill="#fafafa" />
    <rect x="43.533" y="7.5" width="0.5" height="5" stroke="#fafafa" />
    <path d="M44.033 6.5 l-3 3v1 l3 3" stroke="d5d5d5" fill="#fafafa" />
  </g>
  <g fill="#333" text-anchor="middle" font-family="Helvetica Neue,Helvetica,Arial,sans-serif" font-weight="700" font-size="110px" line-height="14px">
    <text x="187.665" y="150" fill="#fff" transform="scale(0.1)" textLength="275.33" lengthAdjust="spacing">Hello</text>
    <text x="187.665" y="140" transform="scale(0.1)" textLength="275.33" lengthAdjust="spacing">Hello</text>
    <text x="625.925" y="150" fill="#fff" transform="scale(0.1)" textLength="301.19" lengthAdjust="spacing">World</text>
    <a target="_blank" xlink:href="http://github.com">
      <text id="rlink" x="625.925" y="140" transform="scale(0.1)" textLength="301.19" lengthAdjust="spacing">World</text>
    </a>
  </g>
  <a target="_blank" xlink:href="http://example.com">
    <rect id="llink" stroke="#d5d5d5" fill="url(#a)" x="0.5" y="0.5" width="37.533" height="19" rx="2" />
  </a>
</svg>
            <h1>Parameters:</h1>          
              <table>
                <thead>
                  <tr> <td>Name</td> <td>Value</td> </tr>
                </thead>
                <tbody>
                  <tr> <td>Style</td> <td>Social</td> </tr>
                  <tr> <td>LeftText</td> <td>Hello</td> </tr>
                  <tr> <td>RightText</td> <td>World</td> </tr>
                  <tr> <td>Color</td> <td>red</td> </tr>
                  <tr> <td>LabelColor</td> <td></td> </tr>
                  <tr> <td>LeftLink</td> <td>http://example.com</td> </tr>
                  <tr> <td>RightLink</td> <td>http://github.com</td> </tr>
                </tbody>
              </table>

            </body>
            </html> 